### DISCUSSION TITLE
Is PAT authentication supported for NuGet aliases?

### DISCUSSION THREAD

**Discussion by fleed** [cake-build/cake#3264] [Q&A] [answered]
Is `ApiKey` the only supported authentication method for NuGet?

I'd like to use `cake` with `VSTS` package management [(https://www.visualstudio.com/en-us/docs/package/nuget/auth)](https://www.visualstudio.com/en-us/docs/package/nuget/auth), but I'd need a `username` + `password` authentication to use the generated `PAT`, while the API seems supporting the `ApiKey` method only.

Any help in the right direction is appreciated.

Thank you

**Comment by devlead** [comment, marked as answer]
@fleed yes it is, you specify is as password when using the [NuGetAddSource(string name, string source, NuGetSourcesSettings settings)](https://cakebuild.net/api/Cake.Common.Tools.NuGet/NuGetAliases/8B9ADF68) alias.
